pandas/core/arrays/datetimelike.py

```diff
@@ -776,31 +768,9 @@ def isin(self, values: ArrayLike) -> npt.NDArray[np.bool_]:
                 if values.dtype != object:
                     return self.isin(values)
 
-                inferred = lib.infer_dtype(values, skipna=False)
-                if inferred not in inferable:
-                    if inferred == "string":
-                        pass
-
-                    elif "mixed" in inferred:
-                        return isin(self.astype(object), values)
-                    else:
-                        return np.zeros(self.shape, dtype=bool)
-
-            try:
-                values = type(self)._from_sequence(values)
-            except ValueError:
+            if "mixed" in lib.infer_dtype(values, skipna=False):
```

Comment:
@jbrockmendel did you intend to allow the mixed object case? `pandas/tests/indexes/test_datetimelike.py::TestDatetimeLike::test_isin`

Reply:
id be OK with deprecating that, but didn't have that in mind with the deprecation here.  the deprecation here was aimed at avoiding a try/except on _from_sequence.

Reply:
i also suspect we can avoid the "mixed" check and just check for object dtype.

Reply:
Cool. I added a comment to deprecate in the future.

Also nice find. I was able to avoid the `infer_dtype` call